crates/bevy_reflect/src/list.rs

```diff
@@ -11,11 +11,15 @@ use crate::{
 ///
 /// This is a sub-trait of [`Array`] as it implements a [`push`](List::push) function, allowing
 /// it's internal size to grow.
+///
+/// **Attention manual implementors**:
```

Comment:
What do you think of:

```rust
/// An ordered, mutable list of [Reflect] items. This corresponds to types like [`std::vec::Vec`].
///
/// This is a sub-trait of [`Array`] as it implements a [`push`](List::push) function, allowing
/// it's internal size to grow.
///
/// This trait expects index 0 to contain the _first_ or _front_ element.
/// The _last_ or _back_ element must refer to the element with the largest index.
/// These two rules above should be upheld by manual implementors.
```

Also, @devil-ira suggested:

> > Personally I prefer `end` over `back`, but maybe that's just me
> 
> Don't mind personally, but `front`/`back` would match the method names and docs for [`VecDeque`](https://doc.rust-lang.org/std/collections/struct.VecDeque.html) so I think we should go with that.

I think this would make the docs more concise. What do you think of this?

Reply:
Yeah I like that. I personally prefer `first`/`last` (or `start`/`end`), but I won't throw up a fight for those terms haha. We can use `front`/`back` if people find that is clearer 😄

Reply:
Okay, I'll make these changes right away! 😄